### PR TITLE
feat: add command to change the runtime

### PR DESCRIPTION
**Pull request by s1n7ax**

### CODE REVIEW

File: lua/java/api/build.lua [hunk 1]
```diff
@@ -3,15 +3,22 @@ local get_error_handler = require('java.handlers.error')
...
+	local jdtls = require('java.utils.jdtls2')
+	local buf_util = require('java.utils.buffer')
+	local notify = require('java-core.utils.notify')
+
+	is_full_build = type(is_full_build) == 'boolean' and is_full_build or true
 
-			JavaCoreJdtlsClient:new(jdtls())
-				:java_build_workspace(true, buf_util.get_curr_buf())
+	return async(function()
+			JavaCoreJdtlsClient(jdtls()):java_build_workspace(
```
[the remark this answers is not in the document]
Comment (s1n7ax) [reply lua/java/api/build.lua:18]: I don't think we have to do that manually. When the needed commands are registered to `vim.lsp.commands`, `workspace/executeClientCommand` would pick it up and run it.

File: lua/java/commands/init.lua [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+			end
+
+			vim.api.nvim_del_autocmd(id)
+		end
+	end,
+
+	group = vim.api.nvim_create_augroup('JavaCommandReg', {}),
+})
+
+M.commands = {
```
[the remark this answers is not in the document]
Comment (s1n7ax) [reply lua/java/commands/init.lua:25]: These are commands LSP can request to execute. There is probably some actions that causes go to definition action in the process. You can see an example here

https://github.com/nvim-java/nvim-java-refactor/blob/7b841d53ed0ba55857debd0e32ce069b2c17e745/lua/java-refactor/refactor-commands.lua?plain=1#L26-L39

Extract variable is causing a rename symbol client command.

File: lua/java/api/build.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (atm1020) [reply lua/java/api/build.lua:18]: @s1n7ax  Its not really clear to me what you mean. Can you explain it in detail?

[the remark this answers is not in the document]
Comment (s1n7ax) [reply lua/java/api/build.lua:18]: Commands we register in `vim.lsp.commands` is used by language server to do stuff. For instance, `jdtls` does not know how to prompt for a user selection when generating constructors. So, it expect us to have `java.action.generateConstructorsPrompt` client command registered.

https://github.com/atm1020/nvim-java/blob/0930c3b658c274413bb898ce8a38c42b751e1898/lua/java/commands/init.lua?plain=1#L199-L201

So the same way, when dependencies are changed, it automatically rebuild the project when the required client commands are registered (as far as I know). Spring boot extension registers classpath listeners when it's loaded. There are client commands to do that as well.

**Pull request merged**